build_cluster_test.py

```diff
@@ -48,5 +48,35 @@ def test_build(self):
             self.log.info('Grafana Web UI: http://%s:3000',
                           self.monitors.nodes[0].public_ip_address)
 
+    def test_use_public_dns_names(self):
+        """
+        Build a Scylla cluster with params defined in data_dir/scylla.yaml
+        Stop cluster.
+        Replace IPs for seeds, listen_address & broadcast_rpc_address on public dns names
+        The cluster works properly.
+        """
+        self.log.info('DB cluster is: %s', self.db_cluster)
+        for node in self.db_cluster.nodes:
+            node.remoter.run('sudo systemctl stop scylla-server.service')
+            node.remoter.run('sudo systemctl stop scylla-jmx.service')
+            node.wait_db_down()
+
+        addresses= {}
+        for node in self.db_cluster.nodes:
+            addresses[node.private_ip_address]=node._instance.public_dns_name
```

Comment:
Even check what type of DB cluster this is, and skip the test in case it's from a backend where `node._instance` does not have the `.public_dns_name` attribute.